PR: https://github.com/odoo/odoo/pull/178556

From: a53383143f2ab0ca2963bfac663165ec1244c9d3
From: Jitendra Prajapat
Date: 2024-09-05 12:21:45

Breaking data model changes scores: del:4 + add:0, change matches:
-    iface_start_categ_id = fields.Many2one('pos.category', string='Initial Category',
-    start_category = fields.Boolean("Start Category", default=False)
-    pos_iface_start_categ_id = fields.Many2one('pos.category', string='Initial Category', compute='_compute_pos_iface_start_categ_id', readonly=False, store=True)
-    pos_start_category = fields.Boolean(related='pos_config_id.start_category', readonly=False)

Total Changes: 20

[IMP] pos*: remove start category feature from point_of_sale

pos*: point_of_sale, pos_restuarant, pos_self_order

In this commit:
==============
The "Start Category" feature has been removed from the Point of Sale.
This feature is now redundant due to the existing "Restrict Category" feature,
which provides similar functionality by limiting the available categories in POS

Details:
* Removed the "Start Category" feature and its associated code.
* The "Restrict Category" feature now effectively replaces the "Start Category"
  by limiting the selectable categories in POS, making the latter unnecessary.

task-4147445

Related Upgrade PR: https://github.com/odoo/upgrade/pull/6449

closes odoo/odoo#178556

Related: odoo/upgrade#6449
Signed-off-by: Vlad Stroia (vlst) <vlst@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -107,8 +107,6 @@ class PosConfig(models.Model):
     iface_print_skip_screen = fields.Boolean(string='Skip Preview Screen', default=True,
         help='The receipt screen will be skipped if the receipt can be printed automatically.')
     iface_tax_included = fields.Selection([('subtotal', 'Tax-Excluded Price'), ('total', 'Tax-Included Price')], string="Tax Display", default='total', required=True)
-    iface_start_categ_id = fields.Many2one('pos.category', string='Initial Category',
-        help='The point of sale will display this product category by default. If no category is specified, all available products will be shown.')
     iface_available_categ_ids = fields.Many2many('pos.category', string='Available PoS Product Categories',
         help='The point of sale will only display products which are within one of the selected category trees. If no category is specified, all available products will be shown')
     customer_display_type = fields.Selection(selection=lambda self: self._get_customer_display_types(), string='Customer Facing Display', help="Show checkout to customers.", default='local')
@@ -158,7 +156,6 @@ class PosConfig(models.Model):
     default_bill_ids = fields.Many2many('pos.bill', string="Coins/Bills")
     use_pricelist = fields.Boolean("Use a pricelist.")
     tax_regime_selection = fields.Boolean("Tax Regime Selection value")
-    start_category = fields.Boolean("Start Category", default=False)
     limit_categories = fields.Boolean("Restrict Categories")
     module_pos_restaurant = fields.Boolean("Is a Bar/Restaurant")
     module_pos_discount = fields.Boolean("Global Discounts")
@@ -334,13 +331,6 @@ class PosConfig(models.Model):
             if config.invoice_journal_id.currency_id and config.invoice_journal_id.currency_id != config.currency_id:
                 raise ValidationError(_("The invoice journal must be in the same currency as the Sales Journal or the company currency if that is not set."))
 
-    @api.constrains('iface_start_categ_id', 'iface_available_categ_ids')
-    def _check_start_categ(self):
-        for config in self:
-            allowed_categ_ids = config.iface_available_categ_ids or self.env['pos.category'].search([])
-            if config.iface_start_categ_id and config.iface_start_categ_id not in allowed_categ_ids:
-                raise ValidationError(_("Start category should belong in the available categories."))
-
     def _check_payment_method_ids(self):
         self.ensure_one()
         if not self.payment_method_ids:

--- a/addons/point_of_sale/models/res_config_settings.py
+++ b/addons/point_of_sale/models/res_config_settings.py
@@ -73,7 +73,6 @@ class ResConfigSettings(models.TransientModel):
     pos_iface_print_skip_screen = fields.Boolean(related='pos_config_id.iface_print_skip_screen', readonly=False)
     pos_iface_print_via_proxy = fields.Boolean(string='Print via Proxy', compute='_compute_pos_iface_print_via_proxy', readonly=False, store=True)
     pos_iface_scan_via_proxy = fields.Boolean(string='Scan via Proxy', compute='_compute_pos_iface_scan_via_proxy', readonly=False, store=True)
-    pos_iface_start_categ_id = fields.Many2one('pos.category', string='Initial Category', compute='_compute_pos_iface_start_categ_id', readonly=False, store=True)
     pos_iface_tax_included = fields.Selection(related='pos_config_id.iface_tax_included', readonly=False)
     pos_iface_tipproduct = fields.Boolean(related='pos_config_id.iface_tipproduct', readonly=False)
     pos_invoice_journal_id = fields.Many2one(related='pos_config_id.invoice_journal_id', readonly=False)
@@ -99,7 +98,6 @@ class ResConfigSettings(models.TransientModel):
     pos_sequence_id = fields.Many2one(related='pos_config_id.sequence_id')
     pos_set_maximum_difference = fields.Boolean(related='pos_config_id.set_maximum_difference', readonly=False)
     pos_ship_later = fields.Boolean(related='pos_config_id.ship_later', readonly=False)
-    pos_start_category = fields.Boolean(related='pos_config_id.start_category', readonly=False)
     pos_tax_regime_selection = fields.Boolean(related='pos_config_id.tax_regime_selection', readonly=False)
     pos_tip_product_id = fields.Many2one('product.product', string='Tip Product', compute='_compute_pos_tip_product_id', readonly=False, store=True)
     pos_use_pricelist = fields.Boolean(related='pos_config_id.use_pricelist', readonly=False)
@@ -212,14 +210,6 @@ class ResConfigSettings(models.TransientModel):
             else:
                 res_config.pos_iface_available_categ_ids = res_config.pos_config_id.iface_available_categ_ids
 
-    @api.depends('pos_start_category', 'pos_config_id')
-    def _compute_pos_iface_start_categ_id(self):
-        for res_config in self:
-            if not res_config.pos_start_category:
-                res_config.pos_iface_start_categ_id = False
-            else:
-                res_config.pos_iface_start_categ_id = res_config.pos_config_id.iface_start_categ_id
-
     @api.depends('pos_iface_available_categ_ids')
     def _compute_pos_selectable_categ_ids(self):
         for res_config in self:
